processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByMergingQueryRunnerV2.java

```diff
@@ -129,8 +129,9 @@ public GroupByMergingQueryRunnerV2(
       combiningAggregatorFactories[i] = query.getAggregatorSpecs().get(i).getCombiningFactory();
     }
 
+    final String queryTmpDir = querySpecificConfig.getTmpDir();
     final File temporaryStorageDirectory = new File(
-        System.getProperty("java.io.tmpdir"),
+        queryTmpDir != null ? queryTmpDir : System.getProperty("java.io.tmpdir"),
```

Comment:
Would be better to have this default populated in the config object's "getTmpDir" method, so any other engines that need scratch space would automatically agree on the default location.